### ISSUE TITLE
Consider a `suppressed_tokens` arg in `generate()`

### ISSUE THREAD

**Issue by abheesht17** [keras-team/keras-nlp#975]
The user can optionally pass a `suppressed_tokens` arg to GPT-2/BART's `generate()` function. We will set the probability of these tokens to 0 (i.e., logits to -infinity), so that they aren't generated. This is super-useful in order to avoid generating a special token (like bos token) during generation.

**Comment by mattdangerw** [comment]
I think the general problem we need to think about is how we  transform logit outputs. We need to be able to ship default transformations (e.g. for whisper logit supression), and potentially allow users a hook to provide there own.

One option is adding more and more sampler config https://github.com/keras-team/keras-nlp/pull/978. A few other options to consider...

 - Users can pass in an arbitrary logit transformation somewhere. `logit_transform_fn=None`.
 - Users need to subclass a language model and override a method to do custom logit transformation.

**Comment by jbischof** [comment]
I do like keeping this general instead of making framework-y promises that we can fix the "bad words"
